modules/services/git-workspace.nix

```
@@ -0,0 +1,153 @@
+          };
+        };
+      });
+    };
+  };
+in {
+  meta.maintainers = [ lib.maintainers.aciceri ];
+  options.services.git-workspace = {
+    enable = lib.mkEnableOption "git-workspace systemd timer";
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = if config.programs.git-workspace.enable then
+        config.programs.git-workspace.package
+      else
+        pkgs.git-workspace;
+      defaultText = lib.literalExpression ''
+        if config.programs.git-workspace.enable then
+          config.programs.git-workspace.package
+        else
+          pkgs.git-workspace;
+      '';
+      description = "The git-workspace package to use";
+    };
+    frequency = lib.mkOption {
+      type = lib.types.str;
+      example = "daily";
+      default = "daily";
+      description = ''
+        The refresh frequency. Check <literal>man systemd.time</literal> for
+        more information on the syntax.
+      '';
+    };
+    environmentFile = lib.mkOption {
+      type = lib.types.path;
+      description = ''
+        Path to the systemd <literal>EnvironmentFile</literal> to keep tokens,
+        do not put a derivation output here since it would expose secrets in
+        the world readable Nix store.
+            </para><para>
+        It contains lines like
+        <literal>
+        GITHUB_TOKEN=ghp_...
+        </literal>
+            </para><para>
+        If you want to use Nix to manage these tokens use a tool like sops-nix
+        or agenix. This file must be readable by the home-manager user.
+      '';
+      example = "/var/lib/git-workspace-tokens";
+    };
+    workspaces = lib.mkOption {
+      type = lib.types.attrsOf workspaceType;
+      default = { };
+      description = ''
+        1:1 representation of a single <literal>workspace.toml</literal> consumed
+        by git-workspace.
+      '';
+      example = lib.literalExpression ''
+        {
+          personal.provider = [
+            {
+              provider = "github"
+              name = "github_username"
+              path = "''${config.home.homeDirectory}/personal_projects"
+              skip_forks = false;
+            }
+            {
+              provider = "gitlab"
+              name = "gitlab_username"
+              # it's possible to use the same path as previous provider
+              # if there are not name clashes
+              path = "''${config.home.homeDirectory}/personal_projects"
+            }
+          ];
+          work.provider = [
+            {
+              provider = "github"
+              name = "github_org"
+              path = "''${config.home.homeDirectory}/work"
+            }
+          ];
+        };
+      '';
+    };
+  };
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      (lib.hm.assertions.assertPlatform "services.git-workspace" pkgs
+        lib.platforms.linux)
+    ];
+    xdg.configFile = lib.mapAttrs' (workspaceName: workspace: {
```

Can separate the top-level attributes in the config section, i.e.,
```suggestion

    xdg.configFile = lib.mapAttrs' (workspaceName: workspace: {
```

etc.